R/R/step.R

```diff
@@ -31,9 +31,10 @@ step <- function(flow, ..., step, r_function = NULL, foreach = NULL, join = FALS
   }
   if (!is.null(r_function)) {
     function_name <- as.character(substitute(r_function))
-    function_hash <- digest::digest(deparse(r_function), algo = "spookyhash")
+    function_hash <- digest::digest(deparse(r_function), algo = "sha256")
+    trunc_function_hash <- substr(function_hash, 1, 16)
     if (length(function_name) > 1) { # likely an anonymous function
-      function_name <- paste(step, "function", function_hash, sep = "_")
+      function_name <- paste(step, "function", trunc_function_hash, sep = "_")
```

Comment:
@mdneuzerling Can we move L34:35 within the `if` clause?

Reply:
also can we clarify the comments a bit? Something like -- `an anonymous function will be parsed into a list of function arguments and function body, while non-anonymous function is only parsed into a function name as a length-one list`